qt/python/mantidqt/widgets/plotconfigdialog/axestabwidget/test/test_axestabwidgetpresenter.py

```diff
@@ -276,6 +276,39 @@ def test_ax_limits_set_correctly(self):
 
         self.assertEqual(presenter.current_view_props["xlim"], (-1.0, 10.0))
 
+    def test_apply_all_properties_with_errors_notifies_parent_presenter_on_exception(self):
+        canvas_draw_exception = Exception("Exception in canvas.draw")
+
+        def raise_():
+            raise canvas_draw_exception
```

Comment:
This can be removed / simplified to
`mock.Mock(side_effect = Exception("Exception in canvas.draw"))`